//  Copyright 2021 SÃ©bastian Dejonghe
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

syntax = "proto3";

import "google/protobuf/timestamp.proto";

package hello_world;

service GrpcServer {
	rpc Login(LoginRequest) returns (LoginResponse) {}

	rpc HelloWorld(HelloWorldRequest) returns (HelloWorldResponse) {}
}

message AuthenticationToken {
	google.protobuf.Timestamp not_after = 1;
	string username = 2;
	repeated string roles = 3;
}

message LoginRequest {
	string username = 1;
	string password = 2;
}

message LoginResponse {
	string auth_token = 1;
}

message HelloWorldRequest {
	string name = 1;
}

message HelloWorldResponse {
    string message = 1;
}